src/content/docs/baas/Firebase/firebase-case-study-AnnaAWojtowicz.md

```diff
@@ -0,0 +1,817 @@
+---
+title: How to use Firebase when creating a website with auctions
+author: Anna Wojtowicz
+tags: Firebase, web app, Cloud Firestore, database, NoSQL, Firebase JS SDK, BaaS, addDoc() method, setDoc() method, updateDoc() method, deleteDoc() method, getDoc() method, getDocs() method
+---
+
+# Case: How to use Firebase when creating a website with auctions.
+
+Firebase is an example of a platform that provides back-end solutions for both front- and back-end developers (in other words a BaaS - Backend as a Service). This platform is especially convenient for front-end developers without back-end skills to build and develop applications.
+
```

Comment:
Will do, sir! :) How much time do I have to do this? I'm asking because I'm away until Sunday and won't be able to write it before that day.

Reply:
Hi! let me know what do you think about this part: 

Alternatives to Firebase
While Firebase remains a leader in Backend as a Service (BaaS), there are several other platforms available on the market, such as AWS Amplify, Supabase, Nhost, and Appwrite. However, according to Google stats as of 10.03.2024, their popularity is nowhere near that of Firebase. Although Firebase leads the way in the BaaS sector, it might be wise to consider exploring other options if the expenses of maintaining a successful app become too high or if Firebase's features do not align completely with the developer's requirements.

AWS Amplify
AWS Amplify, provided by Amazon, focuses on services for both frontend and backend development, offering solutions for full stack development. Similar to Firebase, it provides serverless architecture, API management, real-time database, file storage, and authentication solutions, among others (Amazon, 2024). The fact that AWS Amplify offers full stack development solutions implies that it may be better suited for larger and more complex projects.

Supabase, Nhost, and Appwrite
The main distinction between Firebase and Supabase, Nhost, and Appwrite is that the latter three are open-source platforms. While they offer many features similar to Firebase, such as API management, real-time database, file storage, and authentication solutions, they also provide self-hosting options, offering more control over the infrastructure and differing in pricing. Another differentiating factor is that both Supabase and Nhost use PostgreSQL, while Appwrite uses MongoDB as its default database.
A notable drawback is the fewer features offered by these platforms compared to Firebase (Supabase, 2024; Nhost, 2024; Appwrite, 2024).

Sources: 
Amazon (2024). AWS Simplify. Retrieved 10.03.2024 from: https://aws.amazon.com/amplify/;
Appwrite (2024). Appwrite. Retrieved 10.03.2024 from: https://appwrite.io/; 
MongoDB (2024). MongoDB. Retrieved 10.03.2024 from: https://www.mongodb.com/; 
Nhost (2024). Nhost. Retrieved 10.03.2024 from: https://nhost.io/; 
Supabase (2024). Supabase. Retrieved 10.03.2024 from: https://supabase.com/;